main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+type config struct {
+	RepositoryURL string `env:"repository_url,required"`
+	CloneIntoDir  string `env:"clone_into_dir,required"`
+	Commit        string `env:"commit"`
+	Tag           string `env:"tag"`
+	Branch        string `env:"branch"`
+}
+
+func mainE() error {
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Error: %s\n", err)
+	}
+	stepconf.Print(cfg)
+
+	gitCmd, err := git.New(cfg.CloneIntoDir)
+	if err != nil {
+		return fmt.Errorf("create gitCmd project, error: %v", err)
+	}
+	checkoutArg := getCheckoutArg(cfg.Commit, cfg.Tag, cfg.Branch)
+
+	originPresent, err := isOriginPresent(gitCmd, cfg.CloneIntoDir, cfg.RepositoryURL)
```

Comment:
we do not have use-case for only checking out a new repo state, pls remove this functionality